Assets/Scripts/Scenes/InGame/Enemy/EnemySpawner.cs

```diff
@@ -7,98 +7,218 @@
+        private byte _x, _y,_z;
         [SerializeField]
-        [Tooltip("�}�b�v�̃}�X�ڂ̕�")]
+        [Tooltip("マップのマス目の幅")]
         private float _range;
         [SerializeField]
-        [Tooltip("�ł����E�̒����G�̎��E�̋���")]
+        [Tooltip("最も視界の長い敵の視界の距離")]
         private float _maxVisiviilityRange;
         [SerializeField]
-        [Tooltip("�}�b�v�̃}�X�ڂ̍ł������̃}�X�ڂ̒��S��")]
+        [Tooltip("マップのマス目の最も左下のマス目の中心部")]
         private Vector3 _centerPosition;
 
-        [Header("�쐬����G�̃v���n�u�ꗗ")]
+        [Header("作成する敵のプレハブ一覧")]
         [SerializeField] private GameObject _testEnemy;
         [SerializeField] private GameObject _deepOnes;
         [SerializeField] private GameObject _spawnOfCthulhu;
         [SerializeField] private GameObject _MiGo;
 
 
 
-        [Header("��������ۂ̐ݒ�")]
+        [Header("生成する際の設定")]
         [SerializeField] private Vector3 _enemySpawnPosition;
 
+
+        private List<StageDoor> _doors = new List<StageDoor>();
+
+        private CancellationTokenSource _cancellationTokenSource;
+
         // Start is called before the first frame update
-        void Start()
+        async void Start()
         {
+            _cancellationTokenSource = new CancellationTokenSource();
             if (_nonInGameManagerMode)
             {
-                //�}�b�v���X�L����
-                _enemyVisibilityMap = new EnemyVisibilityMap();
-                _enemyVisibilityMap.debugMode = _debugMode;
-                _enemyVisibilityMap.maxVisivilityRange = _maxVisiviilityRange;
-                _enemyVisibilityMap.GridMake(_x, _z, _range, _centerPosition);
-                _enemyVisibilityMap.MapScan();
 
-                //�e�X�g�Ƃ��Ă�����Enemy������˗����Ă���
-                EnemySpawn(_enemyName, new Vector3(-10, _centerPosition.y + 3, -10));
+                InitialSpawn(_cancellationTokenSource.Token).Forget();
 
             }
             else {
-                IngameManager.Instance.OnPlayerSpawnEvent.Subscribe(_ => InitialSpawn());
+                IngameManager.Instance.OnPlayerSpawnEvent.Subscribe(_ => InitialSpawn(_cancellationTokenSource.Token).Forget());//プレイヤースポーンはマップが完成してから行われる
             }
 
             if (Instance == null)
                 Instance = this;
             else
                 Destroy(this.gameObject);
-
         }
 
-        public void InitialSpawn() {
 
-            //�}�b�v���X�L����
+
+        private async UniTaskVoid InitialSpawn(CancellationToken token) {
+            //ドアを入手
+            foreach (var doorObject in GameObject.FindGameObjectsWithTag("Door").ToArray<GameObject>())
+            {
+                StageDoor getDoorCs;
+                if (doorObject.TryGetComponent<StageDoor>(out getDoorCs))
+                {
+                    _doors.Add(getDoorCs);
+                }
+                else
+                {
+                    Debug.LogWarning("Door.csを持たないDoorタグのオブジェクト「" + doorObject + "」があります");
+                }
+            }

```

Comment:
```suggestion
        private async UniTaskVoid InitialSpawn(CancellationToken token) {
            //ドアを入手
           _doors = FindObjectsOfType<StageDoor>
            }
```
ドアって元々「StageIntract」ってタグが付いてたと思うのですが、できればEnemySpawnerでのドアの取得をタグで行うのではなく、FindObjectsOfTypeでStageDoorスクリプトの取得にはできませんか？

Reply:
提案のやつ駄目そうなので現在考え中。修正次第このコメント消します

Reply:
ドアについているタグは「StageIntract」に戻しておいてくれるとありがたいです

Reply:
プレハブのタグはいじっていないのでご安心を